server/aws-lsp-identity/src/language-server/identityService.ts

```diff
@@ -136,6 +164,66 @@ export class IdentityService {
         }
     }
 
+    async getIamCredential(params: GetIamCredentialParams, token: CancellationToken): Promise<GetIamCredentialResult> {
+        const emitMetric = this.emitMetric.bind(
+            this,
+            'flareIdentity_getIamCredential',
+            this.getIamCredential.name,
+            Date.now()
+        )
+
+        try {
+            const options = { ...getIamCredentialOptionsDefaults, ...params.options }
+
+            token.onCancellationRequested(_ => {
+                if (options.callStsOnInvalidIamCredential) {
```

Comment:
This may apply beyond just cases where AssumeRole is called given the various ways of retrieving credentials.  You should verify with Nik and/or whoever defined metrics for this project to understand the expectations here.